src/actions/clusterActions.js

```diff
@@ -348,7 +349,49 @@ export function clusterInstallApp(app, clusterID) {
       }
     });
 
+    const optionalCreateAppSecret = new Promise((resolve, reject) => {
+      if (Object.keys(app.secretsYAML).length !== 0) {
+        // If we have user config that we want to create, then
+        // fire off the call to create it.
+        appSecretsApi
+          .createClusterAppSecret(clusterID, app.name, {
+            body: app.secretsYAML,
+          })
+          .then(() => {
+            // The call succeeded, resolve the promise
+            resolve();
```

Comment:
Why do you resolve here with no parameters, but on the other `resolve` call, you resolve it with an empty object?

Reply:
If we are only interested in errors, `then()` could be completely removed.